packages/components/src/components/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -0,0 +1,62 @@
+import type { DisclosureState } from "ariakit";
+import React from "react";
+import { Button } from "../Button";
+import { Modal, ModalBody, ModalFooter } from "../Modal/index";
+
+export interface ConfirmationModalProps {
+  children: NonNullable<React.ReactNode>;
+  state: DisclosureState;
+  buttonLabel?: string;
+  buttonVariant?: "destructive" | "primary";
+  onConfirm: () => void;
+}
+
+/**
+ * A modal dialog component that forces the user to acknowledge and make a choice
+ * @param props Object with props
+ * @param props.state state to open and close the modal
+ * @param props.buttonVariant variant of the confirm button
+ * @param props.buttonLabel label for the confirm button
+ * @param props.onConfirm function to call when clicking the confirm button
+ * @param props.children React children for the modal body
+ * @returns React.JSX.Element
+ */
+
+const ConfirmationModal = React.forwardRef<
+  HTMLDivElement,
+  ConfirmationModalProps
+>(
+  (
+    {
+      state,
+      buttonVariant = "primary",
```

Comment:
Inspired by [atomic design](https://bradfrost.com/blog/post/atomic-web-design/#molecules), a beneficial framework for design systems, I prefer to have a version customized for the component, considering that multiple aspects of it may undergo changes.

Currently, the focus is on the button, but it's important to note that this component is part of a larger structure that should change cohesively. We should avoid controlling specific components through props and, instead, ensure that the output maintains consistency for combinations that require it.